src/main/java/io/aiven/kafka/connect/opensearch/OpensearchSinkTask.java

```diff
@@ -157,8 +157,10 @@ protected String convertTopicToIndexName(final String topic) {
 
     private void ensureIndexExists(final String index) {
         if (!indexCache.contains(index)) {
-            LOGGER.info("Create index {}", index);
-            client.createIndex(index);
+            if (!client.indexExists(index)) {
```

Comment:
@LucasStinchcombe thanks for the fix, but it won't address the problem at large: there are still posibility for index to be created in between of `indexExists` and `createIndex` calls.  I think we should be focusing on the fact that index may exists during the `createIndex`: should that be fatal or non-fatal condition?

Reply:
Sure, if we want it to be non-fatal at `createIndex`, then following suite with current implementation, we would just ignore two additional cases in [createIndex:L124](https://github.com/aiven/opensearch-connector-for-apache-kafka/blob/main/src/main/java/io/aiven/kafka/connect/opensearch/OpensearchClient.java#L124). Although I personally think that checking strings in return error is brittle. I'll check what the values of `RESOURCE_EXISTS_AS_ALIAS` and `RESOURCE_EXISTS_AS_DATASTREAM` should be.

```java
if (!(e.getMessage().contains(RESOURCE_ALREADY_EXISTS_EXCEPTION)
    | | e.getMessage().contains(RESOURCE_EXISTS_AS_ALIAS)
    | | e.getMessage().contains(RESOURCE_EXISTS_AS_DATASTREAM)))
{
  throw e;
}
return false;
```

Let me know what you think.

Reply:
:+1:  Thanks @LucasStinchcombe , I think it would be concise with the way the error handling is implemented right now (even if it is not ideal), thanks again.